kong/tracing/instrumentation.lua

```diff
@@ -204,7 +209,7 @@ function _M.request(ctx)
       ["http.url"] = req_uri,
       ["http.host"] = host,
       ["http.scheme"] = scheme,
-      ["http.flavor"] = ngx.req.http_version(),
+      ["http.flavor"] = http_flavor,
       ["net.peer.ip"] = client.get_ip(),
```

Comment:
Would [http.client_ip](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions) be more appropriate than `net.peer.ip` to identify the client ip here? I am just going by the linked docs, if there is any reason why we are using this name please disregard my comment.

Reply:
Maybe @mayocream knows why this name was used.

Reply:
Good question.

The `http.client_ip` attribute is described like this:

> The IP address of the original client behind all proxies, if known (e.g. from X-Forwarded-For)

Operative word is *original*.

What `kong.client.get_ip()` produces is explicitly *not the original*. It says so in the [docs](https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.client/#kongclientget_ip):

> This module always returns the address of the client directly connecting to Kong. That is, **in cases when a load balancer is in front of Kong, this function returns the load balancer’s address, and not that of the downstream client
**.

If we wanted to set "the original" then our best bet would be `kong.client.get_forwarded_ip()` instead:

```
["http.client_ip"] = client.get_forwarded_ip()
```

But I am not surprised that @mayocream didn't include that one because it is mired with conditionals (*balancer is one of the trusted IPs, and the load balancer adds the right headers matching with the configuration of `real_ip_header`*)

Anyway, out of the scope for this PR. I already overexceeded it by including http.flavor

Reply:
But now that I look at it more closely, it seems like the attribute `net.peer.ip` isn't part of the spec ... but `net.sock.peer.addr` is. I will leave that one for the future.